p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/repository/P2RepositoryManager.java

```diff
@@ -153,11 +211,18 @@ private IMetadataRepository getMetadataRepositor(URI location, String id) throws
 		return metadataManager.loadRepository(location, new LoggerProgressMonitor(logger));
 	}
 
-	public void downloadArtifact(IInstallableUnit iu, IArtifactRepository artifactRepository,
-			OutputStream outputStream) throws IOException {
+	public static void assertNoError(IStatus status, Path targetLocation) throws ProvisionException {
+		if (!status.isOK() && status.matches(IStatus.ERROR)) {
+			String msg = "Failed to create output repository at: " + targetLocation + "\n";
+			throw new ProvisionException(msg + StatusTool.toLogMessage(status), StatusTool.findException(status));
```

Comment:
This is still wrong, `toLogMessage` (as the name suggest) is when output it as a log message not using it as an exception message.

I would also not do to much unwrapping here, so why not simply create an multi error status and attach the original status as a child? Or even throw a ProvisionException with the status itself, I don't think the message here buys us much as the status should already include why it failed.